### ISSUE TITLE
Attention Updates

### ISSUE THREAD

**Issue by dsikka** [vllm-project/speculators#254]
The key difference from Eagle3 is that DFlash uses a non-causal $[L, 2L]$ attention pattern where queries attend to both the hidden states from the verifier (first $L$ positions) and the mask token embeddings (last $L$ positions), all in a single forward pass. 

Create the attention mask modifiers for flash attention. The attention mask within a document is demonstrated in the [RFC](https://github.com/vllm-project/speculators/issues/248), but we also need to mask out other documents when we concatenate the training data.

See further details: https://github.com/vllm-project/speculators/issues/248

**Comment by shanjiaz** [comment]
Hi @Vishnu-sai-teja Thanks so much for reaching out! @MeganEFlynn is currently working on this as part of the training support effort. We'll definitely reach out when we have a more well-defined task for you! Meanwhile, let us know if you have any questions on issue [259](https://github.com/vllm-project/speculators/issues/259).
